### DISCUSSION TITLE
scroll restoration without scroll animation

### DISCUSSION THREAD

**Discussion by vineetdigit** [TanStack/query#3510] [Q&A]
I'm wondering if there is a way to disable the scroll animation with scroll restoration? 

For example, if I have a long list of posts. I scroll down to one of the posts and click on it. Later when I navigate using back button, the position is restored to the post that I clicked. But it does so by scrolling automatically all the way down to this post. What I'm expecting is something equivalent of 

```
window.scrollTo({
  top: <some offset>,
  left: 0,
  behavior: 'instant',
});
```
where upon navigating back, the page is already positioned at the right offset without the scroll effect.

Is this possible?

**Comment by TkDodo** [comment]
react-query does nothing for scroll restoration - it's the browser doing it. https://developer.mozilla.org/en-US/docs/Web/API/History/scrollRestoration

**Comment by vineetdigit** [comment]
Right. And I read another reply in which you mentioned this. Yet this page https://react-query.tanstack.com/guides/scroll-restoration made me think it's react-query that's adding scroll restoration. 

I know this question is no longer related to RQ, but please help me understand this while we're at it. Is the scroll restoration with animation a default behavior of Chrome? 

I don't believe I have explicitly enabled scroll restoration anywhere in my react app. So my guess is that it must be one of the libraries that I'm using. I suspected it was RQ because of the above documentation. I don't think even React Router is adding it.

**Reply by TkDodo** [reply]
if you want, you can make a PR to the docs saying that it just works ™️  because the browser does it :)

can you show this behaviour in a sandbox? we can then test it in multiple browsers to see if there are differences.

**Reply by TkDodo** [reply]
i can actually not get scroll restoration to work in any of our examples. maybe its a codesandbox thing, but I always stay at the top of the page :/

**Comment by vineetdigit** [comment]
That's interesting. I will continue to investigate where it got added.
Really appreciate your answers. Thanks Dominik!
